includes/admin/class-wc-rest-stripe-settings-controller.php

```diff
@@ -605,22 +604,6 @@ private function update_enabled_payment_methods( $payment_method_ids_to_enable,
 			return;
 		}
 
-		if ( ! $is_upe_enabled ) {
-			$currently_enabled_payment_method_ids = WC_Stripe_Helper::get_legacy_enabled_payment_method_ids();
-			$payment_gateways                     = WC_Stripe_Helper::get_legacy_payment_methods();
-
-			foreach ( $payment_gateways as $gateway ) {
-				$gateway_id = str_replace( 'stripe_', '', $gateway->id );
-				if ( ! in_array( $gateway_id, $payment_method_ids_to_enable, true ) && in_array( $gateway_id, $currently_enabled_payment_method_ids, true ) ) {
-					$gateway->update_option( 'enabled', 'no' );
-				} elseif ( in_array( $gateway_id, $payment_method_ids_to_enable, true ) ) {
-					$gateway->update_option( 'enabled', 'yes' );
-				}
-			}
-
-			return;
-		}
-
 		if ( $this->gateway instanceof WC_Stripe_UPE_Payment_Gateway ) {
 			$this->gateway->update_enabled_payment_methods( $payment_method_ids_to_enable );
 		}
```

Comment:
Is this condition needed? I don't think we create a non-UPE instance any more.
```suggestion
		$this->gateway->update_enabled_payment_methods( $payment_method_ids_to_enable );
```